diff --git a/src/errno/__errno_location.c b/src/errno/__errno_location.c
index 7f9d602..42cb11d 100644
--- a/src/errno/__errno_location.c
+++ b/src/errno/__errno_location.c
@@ -1,9 +1,11 @@
 #include <errno.h>
 #include "pthread_impl.h"
 
+static int __errno_value = 0;
+
 int *__errno_location(void)
 {
-       return &__pthread_self()->errno_val;
+       return &__errno_value;
 }
 
 weak_alias(__errno_location, ___errno_location);
